module/qr_code.py

```diff
@@ -11,45 +168,46 @@ def __init__(self, code):
         self.data = None
         self.type = code.type
         self.rect = code.rect
-        self.polygon = code.polygon
-        self.distance = None
+        self.polygon = QrPolygon(code.polygon)
+        self.center_distance = None
         self.center_offset = list()
         self.center_offset_mm = list()
 
         self.parse_data(code.data.decode("utf-8"))
 
-    def calculate_distance(self, camera_properties, frame_height):
+    def calculate_center_distance(self, camera_properties, frame_height):
         """ Calculate the distance between the QR code and the camera.
         :param camera_properties: Properties of the camera used in making the frame/picture.
         :param frame_height: Height of the frame/picture in which the QR code is found.
         :return: Nothing.
         """
         try:
             distance_to_object = (camera_properties.get_focal_length() * int(self.get_value("Height"))
-                                  * frame_height) / (self.rect.height * camera_properties.get_sensor_height())
-            self.distance = distance_to_object
+                                  * frame_height) / (self.polygon.middle_height * camera_properties.get_sensor_height())
+            self.center_distance = distance_to_object
         # Sometimes something goes wrong in calculating the distance
         # resulting in an ZeroDivisionError.
         # If this is the case nothing should happen.
         # Else the program will crash.
         except ZeroDivisionError:
-            self.distance = None
+            self.center_distance = None
 
     def calculate_center_offset(self, frame_width, frame_height):
         """ Calculates the offset of the QR code relative to the center of the frame/image.
         :param frame_width: Width of the frame/image.
         :param frame_height: Height of the frame/image.
         :return: Nothing.
         """
-        qr_center = self.get_qr_center()
-        offset_x, offset_y, = qr_center[0] - \
-            frame_width/2, qr_center[1] - frame_height/2
+        qr_center = self.polygon.center_point
+        offset_x, offset_y, = qr_center.x - frame_width/2, qr_center.y - frame_height/2
```

Comment:
Please seperate these calculations into a single line for each. It could easily be missed that there are two different calculations being made here.

Reply:
I have done so in lines 229 and 230 in commit https://github.com/R2D2-2019/vision/pull/11/commits/2d4a20c0fd93867a7d668319a7c45ef5a00d22f7